src/types.ts

```diff
@@ -83,10 +83,33 @@ export interface MessageParams {
   spec: SchemaSpec<any> & Record<string, unknown>;
 }
 
+/**
+ * Placeholder to allow adding to the message union by type augmentation
+ *
+ * @example to add `ReactElement` to allowed message types
+ * ```ts
+ * declare module 'yup/lib/types' {
+ *   export interface CustomMessage extends BaseMessage {
+ *     type: ReactElement;
+ *   }
+ * }
+ * ```
+ * NB types cannot be removed from the message union with this, only added
+ */
+export interface CustomMessage {}
+export interface BaseMessage {
+  type: unknown;
```

Comment:
question: why is this approach better, vs extending the `Message` interface to be the type you want?

Reply:
As far as I know, you can't extend a union declared in another library (whereas you can augment interfaces, which this does do)

We (as a consumer) could have a wrapper around yup which defined `type OurMessage = Message | ReactElement`, but then we'd have to override all the yup types to accept it everywhere that yup's types use `Message` (or wrap all of yup etc).